### ISSUE TITLE
[Super Cache] Deprecation notice on PHP 8.1

### ISSUE THREAD

**Issue by dilirity** [Automattic/jetpack#29971]
### Impacted plugin

Super Cache

### Quick summary

A user reported a deprecation warning when running Super Cache on PHP 8.1 - https://wordpress.org/support/topic/compatibility-with-php-8-1-7/

### Steps to reproduce

TBD

### A clear and concise description of what you expected to happen.

There should be no notice.

### What actually happened

There is a notice.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

**Comment by mcaskill** [comment]
I've encountered this issue on my end.

I've patched `wp_cron_preload_cache()` in my project with the following:

```diff
 	if ( $page_on_front != 0 && ( $post_id == $page_on_front || $post_id == $page_for_posts ) )
 		continue;
 	$url = get_permalink( $post_id );
-
+	if ( !is_string( $url ) ) {
+		wp_cache_debug( "wp_cron_preload_cache: skipped $post_id. Expected a URL, received: " . gettype( $url ) );
+		continue;
+	}
 	if ( wp_cache_is_rejected( $url ) ) {
 		wp_cache_debug( "wp_cron_preload_cache: skipped $url per rejected strings setting" );
 		continue;
```

The `get_permalink()` function is documented as returning either `string` or `false` but with its various filters, could return anything.

I hope this helps.

Unfortunately, I don't have time to create a pull request.

**Comment by bobmatyas** [comment]
Another user shared their solution on:

https://wordpress.org/support/topic/compatibility-with-php-8-1-7/#post-16926772

**Comment by haqadn** [comment]
Although related, this doesn't seem to be a duplicate of #32336

**Comment by donnchawp** [comment]
This is fixed in trunk, so the next release will have it. I'll leave this open because @mcaskill reported a different issue I'll fix.

**Comment by donnchawp** [comment]
The issue @mcaskill reported has been fixed in #32484
